python_testcontainers/infrahub_testcontainers/container.py

```diff
@@ -298,26 +283,120 @@ def database_restore_backup(self, backup_file: Path) -> None:
+
+            # Waiting for cluster to stabilize...
+            time.sleep(10)
+
+            self.exec_in_container(
+                command=["cypher-shell", "-u", "neo4j", "-p", "admin", "DROP DATABASE neo4j;"],
+                service_name=service_name,
+            )
+
+            self.exec_in_container(
+                command=["rm", "-rf", "/data/databases/neo4j"],
+                service_name=service_name,
+            )
+            self.exec_in_container(
+                command=["rm", "-rf", "/data/transactions/neo4j"],
                 service_name=service_name,
             )
-            raise Exception(f"Failed to restore database:\n{restore_output}\nDebug logs:\n{debug_logs}")
 
-        old_services = self.services
-        self.services = ["infrahub-server", "task-worker"]
-        self.stop(down=False)
-        try:
+            self.exec_in_container(
+                command=[
+                    "neo4j-admin", "database", "restore",
+                    "--from-path", str(self.internal_backup_dir / backup_file.name),
+                    "neo4j",
+                ],
+                service_name=service_name,
+            )
+
+            cmd = self.compose_command_property[:]
+            cmd += ["restart", "database"]
+            self._run_command(cmd=cmd)
+            # Waiting for cluster to stabilize...
+            time.sleep(10)
+
+            server_id = None
+            try:
+                stdout, _, _ = self.exec_in_container(
+                    command=[
+                        "cypher-shell", "-d", "system", "-u", "neo4j", "-p", "admin",
+                        'SHOW SERVERS YIELD name, address WHERE address = "database:7687" RETURN name;'
+                    ],
+                    service_name=service_name,
+                )
+                print(f"Server ID query result: {stdout}")
+
+                lines = [line.strip() for line in stdout.splitlines() if line.strip() and not line.startswith("+")]
+                for line in lines:
+                    if line.startswith('"') and line.endswith('"'):
+                        server_id = line.strip('"')
+                        break
+            except Exception as e:
+                print(f"Error retrieving server ID with direct query: {e}")
+
+            if server_id:
+                print(f"Using server ID: {server_id} as seeder")
+                self.exec_in_container(
+                    command=[
+                        "cypher-shell", "-d", "system", "-u", "neo4j", "-p", "admin",
+                        f"CREATE DATABASE neo4j TOPOLOGY 3 PRIMARIES OPTIONS {{ existingData: 'use', existingDataSeedInstance: '{server_id}' }};"
+                    ],
+                    service_name=service_name,
+                )
+
+            print("Checking database status...")
+            self.exec_in_container(
+                command=["cypher-shell", "-d", "system", "-u", "neo4j", "-p", "admin", "SHOW DATABASE neo4j;"],
```

Comment:
This is not enough, you should run the check if database is online loop (like the loop above)